### PR TITLE
Support delegated admin

### PR DESCRIPTION
**Pull request by cdammanintopix**
Since Nextcloud 30.0.0, we can delegate users administration to non-admin groups.
This pull request adds this feature support in the impersonate app

### CODE REVIEW

**Comment by cdammanintopix** [comment]
@blizzz could you look at this? 🙂

**Comment by blizzz** [comment]
@cdammanintopix Thanks for your contribution! Have you tried it though? The change requires to be both admin and delegated admin.

**Comment by blizzz** [comment]
> Yes, I tested it with both an admin account and with a delegated admin account

Ah, yes, makes sense, got tricked by the negation :see_no_evil: Sorry for that.

Could you run `composer i && composer run cs:fix` and amend the changes? It's for code style compliance.

File: lib/Controller/SettingsController.php [hunk 1]
```diff
@@ -74,6 +74,7 @@ public function impersonate(string $userId): JSONResponse {
 		}
 
 		if (!$this->groupManager->isAdmin($impersonator->getUID())
+				&& !$this->groupManager->isDelegatedAdmin($impersonator->getUID())
```
Comment (blizzz) [inline lib/Controller/SettingsController.php:77]:
```suggestion
			&& !$this->groupManager->isDelegatedAdmin($impersonator->getUID())
```

just one tab less :)

**Review by blizzz — COMMENTED** [review]
sorry, have to nitpick once more :bow:

**Review by blizzz — APPROVED** [review]

**Comment by cdammanintopix** [comment]
@blizzz what are the next steps for this? Once merged, it will be automatically deployed as an app update in Nextcloud portal?

**Comment by blizzz** [comment]
> @blizzz what are the next steps for this? Once merged, it will be automatically deployed as an app update in Nextcloud portal?

Not automatically, we still have to trigger it.

**Pull request merged**